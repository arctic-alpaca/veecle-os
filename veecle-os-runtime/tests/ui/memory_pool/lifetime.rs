use veecle_os_runtime::memory_pool::MemoryPool;

fn main() {
    let _chunk = {
        let pool: MemoryPool<usize, 2> = MemoryPool::new();
        pool.chunk(0)
    };
}
