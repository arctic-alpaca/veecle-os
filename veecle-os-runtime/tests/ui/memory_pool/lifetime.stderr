error[E0597]: `pool` does not live long enough
 --> tests/ui/memory_pool/lifetime.rs:6:9
  |
4 |     let _chunk = {
  |         ------ borrow later stored here
5 |         let pool: MemoryPool<usize, 2> = MemoryPool::new();
  |             ---- binding `pool` declared here
6 |         pool.chunk(0)
  |         ^^^^ borrowed value does not live long enough
7 |     };
  |     - `pool` dropped here while still borrowed
